src/System.Private.DataContractSerialization/src/System/Xml/ValueHandle.cs

```diff
@@ -877,10 +868,8 @@ private string GetCharText()
             if (ch > char.MaxValue)
             {
                 SurrogateChar surrogate = new SurrogateChar(ch);
-                char[] chars = new char[2];
-                chars[0] = surrogate.HighChar;
-                chars[1] = surrogate.LowChar;
-                return new string(chars, 0, 2);
+                Span<char> chars = stackalloc char[2] { surrogate.HighChar, surrogate.LowChar };
```

Comment:
This entire method can be implemented as a one-liner:

```cs
private string GetCharText() => char.ConvertFromUtf32(GetChar());
```

Reply:
You are right! I made the appropriate change in https://github.com/dotnet/corefx/pull/36838/commits/1daf838d3f5a474856e9c3503604ba37758ad8fc

Reply:
Tests seem to be failing for the one-liner. The SurrogateChar constructor is actually doing some work to transform the parameter. I am going to revert back to the previous implementation.

https://github.com/dotnet/corefx/blob/master/src/System.Private.DataContractSerialization/src/System/Text/SurrogateChar.cs#L24-L33